Assets/Scripts/AudioManager.cs

```diff
@@ -0,0 +1,88 @@
+﻿using UnityEngine.Audio; 
+using UnityEngine;
+using System;
+
+public class AudioManager : MonoBehaviour
+{
+    public static AudioManager instance { get; private set; }
+    public Sound[] musicTracks;
+    public Sound[] sfxTracks;
+    
+
+    private void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+            return; 
+        }
+
+        foreach(Sound music in musicTracks)
+        {
+            music.source = gameObject.AddComponent<AudioSource>(); 
+            music.source.clip = music.audioClip;
+            music.source.loop = music.loop;
+            music.source.volume = music.volume;
+            music.source.pitch = music.pitch; 
+        }
+
+        foreach(Sound sfx in sfxTracks)
+        {
+            sfx.source = gameObject.AddComponent<AudioSource>();
+            sfx.source.clip = sfx.audioClip;
+            sfx.source.loop = sfx.loop;
+            sfx.source.volume = sfx.volume;
+            sfx.source.pitch = sfx.pitch;
+        }
+    }
+
+    public void AdjustVolume(float newVolume)
+    {
+        AudioListener.volume = newVolume;
+    }
```

Comment:
This adjusts in instantly, so if you went from max to min, it would be very jarring. I don't know if this is better, but for a smooth volume roll off, I suggest you create a coroutine that does so...


`public void AdjustVolume(float newVolume) {
    StartCoroutine(AdjustVolumeHelper(newVolume, 1f));
}

private IEnumerator AdjustVolumeHelper(float newVolume, float lerpSpeed) {
    while (AudioListener.volume != newVolume) {
        // Lerps the Master Volume to the new Volume
        // LerpSpeed affects how fast the volume changes
        AudioListener.volume = Mathf.Lerp(AudioListener.volume, newVolume, Time.deltaTime * lerpSpeed);

        // Makes it so the Loop isn't infinite. Play with the floats to change the volume clip boundary.
        if (AudioListener.volume > newVolume - 0.1f || AudioListener.volume < newVolume + 0.1f) { 
            AudioListener.volume = newVolume;
            break;
        }
        yield return null; // Continues in the next frame
    }
}`

Reply:
Done. This will be the next step of my work, when I hook up our pause menu slider and find SFX tracks.